lib/cdo/delete_accounts_helper.rb

```diff
@@ -79,32 +79,34 @@ def clean_survey_responses(user_id)
 
   # Remove all user generated content associated with any PD the user has been through, as well as
   # all PII associated with any PD records.
-  # @param [Integer] The ID of the user to clean the PD content.
-  def clean_and_destroy_pd_content(user_id)
-    PeerReview.where(reviewer_id: user_id).each(&:clear_data)
+  # @param [User] user being purged
+  def clean_and_destroy_pd_content(user)
+    user.workshops_as_facilitator = []
 
-    Pd::Application::ApplicationBase.with_deleted.where(user_id: user_id).each do |application|
+    PeerReview.where(reviewer_id: user.id).each(&:clear_data)
+
+    Pd::Application::ApplicationBase.with_deleted.where(user_id: user.id).each do |application|
       application.form_data = '{}'
       application.notes = nil
       application.save! validate: false
     end
 
     # Two different paths to anonymizing attendance records
-    Pd::Attendance.with_deleted.where(teacher_id: user_id).each do |attendance|
+    Pd::Attendance.with_deleted.where(teacher_id: user.id).each do |attendance|
       attendance.destroy!
       attendance.update!(teacher_id: nil)
     end
-    Pd::Attendance.with_deleted.where(marked_by_user_id: user_id).each do |attendance|
+    Pd::Attendance.with_deleted.where(marked_by_user_id: user.id).each do |attendance|
       attendance.update!(marked_by_user_id: nil)
     end
 
-    Pd::TeacherApplication.where(user_id: user_id).each(&:destroy)
-    Pd::FacilitatorProgramRegistration.where(user_id: user_id).each(&:clear_form_data)
-    Pd::RegionalPartnerProgramRegistration.where(user_id: user_id).each(&:clear_form_data)
-    Pd::WorkshopMaterialOrder.where(user_id: user_id).each(&:clear_data)
-    Pd::InternationalOptIn.where(user_id: user_id).each(&:clear_form_data)
+    Pd::TeacherApplication.where(user_id: user.id).each(&:destroy)
```

Comment:
There's a lot of classes that derive from `Pd::Form`. If there was a way to call `clear_form_data` on all subclasses of `Pd::Form` that would be a lot nicer than this. As of now we'll have to update this every time we make a new class

Reply:
`ObjectSpace.each_object(Class).select {|c| c.included_modules.include? Pd::Form}` should do it